src/init.cpp

```diff
@@ -1104,6 +1105,11 @@ bool AppInitParameterInteraction()
         boost::split(vstrReplacementModes, strReplacementModeList, boost::is_any_of(","));
         fEnableReplacement = (std::find(vstrReplacementModes.begin(), vstrReplacementModes.end(), "fee") != vstrReplacementModes.end());
     }
+    const int64_t replacement_timeout = gArgs.GetArg("-mempoolreplacementtimeout", DEFAULT_REPLACEMENT_TIMEOUT);
+    const int64_t expiry = gArgs.GetArg("-mempoolexpiry", DEFAULT_MEMPOOL_EXPIRY) * 60 * 60;
+    if (replacement_timeout < 0 || replacement_timeout > expiry) {
+            return InitError("mempoolreplacementtimeout has to be a non negative number below or equal to mempoolexpiry (in seconds)");
+    }
 
```

Comment:
Not sure if you might also want to check if it is accidentally(?) set above 6hours and print a warning?

Reply:
It's a good question. Would there be a good reason to set it above 6 hours if 6 hours is the default, especially long term?

Reply:
Probably not, since you'd be potentially censoring yourself from valid and about-to-be-mined txes.

Reply:
Maybe it can be useful in testing or to have old behavior, hence adding a log warning entry seems to me a good idea but i'll maybe wait for others to agree as we don't want to spam the logs either